apis/python/src/tiledb/vector_search/object_readers/directory_reader.py

```diff
@@ -115,39 +344,107 @@ def get_partitions(
         return partitions
 
 
+class TileDBLoader:
+    def __init__(
+        self,
+        uri: str,
+        config: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        from langchain_community.document_loaders.parsers.generic import (
+            MimeTypeBasedParser,
+        )
+        from langchain_community.document_loaders.parsers.html import BS4HTMLParser
+        from langchain_community.document_loaders.parsers.msword import MsWordParser
+        from langchain_community.document_loaders.parsers.pdf import PyMuPDFParser
+        from langchain_community.document_loaders.parsers.txt import TextParser
+
+        self.uri = uri
+        self.config = config
+        self.parser = MimeTypeBasedParser(
+            handlers={
+                "application/pdf": PyMuPDFParser(),
+                "text/plain": TextParser(),
+                "text/html": BS4HTMLParser(),
+                "application/msword": MsWordParser(),
+                "application/vnd.openxmlformats-officedocument.wordprocessingml.document": (
+                    MsWordParser()
+                ),
+            },
+            fallback_parser=None,
+        )
+
+    def load(self) -> List:
+        """Load data into Document objects."""
+        return list(self.lazy_load())
+
+    def lazy_load(
+        self,
+    ) -> Iterator:
+        """A lazy loader for Documents."""
+        import mimetypes
+
+        from langchain_community.document_loaders.blob_loaders import Blob
+
+        vfs = tiledb.VFS(config=self.config)
```

Comment:
Context/scope context are needed for all the TileDB calls here and below.

Reply:
See comment for removing context